src/Symfony/Bridge/Doctrine/HttpFoundation/DbalSessionHandler.php

```diff
@@ -241,9 +242,30 @@ private function getMergeSql()
                     "WHEN MATCHED THEN UPDATE SET $this->dataCol = :data, $this->timeCol = :time;";
             case 'sqlite' === $platform:
                 return "INSERT OR REPLACE INTO $this->table ($this->idCol, $this->dataCol, $this->timeCol) VALUES (:id, :data, :time)";
-            case 'postgresql' === $platform && version_compare($this->con->getServerVersion(), '9.5', '>='):
+            case 'postgresql' === $platform && version_compare($this->getServerVersion(), '9.5', '>='):
                 return "INSERT INTO $this->table ($this->idCol, $this->dataCol, $this->timeCol) VALUES (:id, :data, :time) ".
                     "ON CONFLICT ($this->idCol) DO UPDATE SET ($this->dataCol, $this->timeCol) = (EXCLUDED.$this->dataCol, EXCLUDED.$this->timeCol)";
         }
     }
+
+    private function getServerVersion()
+    {
+        $params = $this->con->getParams();
+
+        if (isset($params['serverVersion'])) {
+            return $params['serverVersion']; // Explicit platform version requested (supersedes auto-detection), so we respect it.
```

Comment:
Nitpicking: these long comments on the side of the code are a bit "strange" and, if I recall it correctly, not frequently used in Symfony. We usually put them just above the commented code.